server/server.go

```diff
@@ -409,15 +413,27 @@ func (s *server) WaitForHealthy(ctx context.Context, _ *rpcpb.WaitForHealthyRequ
 		}
 		if s.clusterInfo.CustomChainsHealthy {
 			defer s.mu.RUnlock()
-			return &rpcpb.WaitForHealthyResponse{ClusterInfo: s.clusterInfo}, nil
+			clusterInfo, err := deepCopy(s.clusterInfo)
+			if err != nil {
+				return nil, err
+			}
+			return &rpcpb.WaitForHealthyResponse{ClusterInfo: clusterInfo}, nil
 		}
 		select {
-		case err := <-s.asyncErrCh:
+		case <-s.asyncErrCh:
```

Comment:
we need to keep this err from asyncErrCh and return it together with WaitForHealthyResponse

Reply:
I mean keep `case asyncErr := <-s.asyncErrCh:` and later on do `return &rpcpb.WaitForHealthyResponse{ClusterInfo: clusterInfo}, asyncErr`

Reply:
I would also combine deepCopy err with asyncErr to return for the deepCopy err case